src/codemodder/result.py

```diff
@@ -176,18 +177,33 @@ def fuzzy_column_match(pos: CodeRange, location: Location) -> bool:
     )
 
 
-class ResultSet(dict[str, dict[Path, list[Result]]]):
-    results_for_rule: dict[str, list[Result]]
+class Rule(CodetfRule):
```

Comment:
@clavedeluna what if instead of being prescriptive about the specific metadata fields here, we instead add a field with an abstract type that conforms to some API like `get_metadata_summary() -> str` and allow each tool to define the way they want to present it?

I also don't want to overengineer this without having a lot of examples to work with.